### PR TITLE
Backwards-compatibility patch for defaultTemperature fix

### PR DESCRIPTION
**Pull request by greentfrapp**
Maps defaultTemperature to F

### CODE REVIEW

File: nlp/compat.js [hunk 1]
```diff
@@ -120,10 +120,22 @@ function deviceNames(locale, result, entities) {
     result.code = newCode;
 }
 
+function defaultUnits(locale, result, entitites) {
+    for (let i = 0; i < result.code.length; i++) {
+        if (result.code[i].startsWith('default')) {
```
Comment (gcampax) [inline nlp/compat.js]: This will run in NN syntax, so the token will be `unit:defaultTemperature`, and it should be replaced with `unit:F`.

**Comment by gcampax** [comment]
You should probably add tests to tests/unit/test_compat.js

File: nlp/compat.js (hunk 1 above)
Comment (greentfrapp) [reply nlp/compat.js]: Fixed!

**Comment by greentfrapp** [comment]
Added tests to test_nlp_compat. @gcampax I also fixed what I think was a bug in `test_nlp_compat.js` that caused only first 4 tests to run. Can you confirm that it was a bug and not intentional?

**Pull request merged**